File: 70. Climbing Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        res = [1,2]
        if(n==1):
            return 1
        
        if(n==2):
            return 2
        
        if(n==3):
            return 3
        
        while len(res)<n:
            res.append(res[-1]+res[-2])
        
        return res[-1]

File: 70. Climbing Stairs.cs
public class Solution {
    public int ClimbStairs(int n) {
        if(n==0||n==1){
            return 1;
        }

        List<int> dp = new List<int>();
        dp.Add(1);
        dp.Add(1);

        for(int i = 2; i <= n; i++){
            dp.Add(dp[i-1] + dp[i-2]);
        }
        return dp[n];
    }
}
